java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -247,4 +252,65 @@ public void set_missing_value_and_returnOldValue_is_null(BaseClient client) {
         String data = client.set("another", ANOTHER_VALUE, options).get();
         assertNull(data);
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void sadd_srem_scard_smembers_existing_set(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        assertEquals(
+                4, client.sadd(key, new String[] {"member1", "member2", "member3", "member4"}).get());
+        assertEquals(1, client.srem(key, new String[] {"member3", "nonExistingMember"}).get());
+
+        Set<String> expectedMembers = Set.of("member1", "member2", "member4");
+        assertEquals(expectedMembers, client.smembers(key).get());
+        assertEquals(1, client.srem(key, new String[] {"member1"}).get());
+        assertEquals(2, client.scard(key).get());
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void srem_scard_smembers_non_existing_key(BaseClient client) {
+        assertEquals(0, client.srem("nonExistingKey", new String[] {"member"}).get());
+        assertEquals(0, client.scard("nonExistingKey").get());
+        assertEquals(Set.of(), client.smembers("nonExistingKey").get());
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void sadd_srem_scard_smembers_key_with_non_set_value(BaseClient client) {
```

Comment:
we dont have to check the error message, we can just check that an error is being raised

Reply:
@shohamazon https://github.com/aws/glide-for-redis/blob/main/python/python/tests/test_async_client.py#L831 most of our IT tests are based on Node and Py and it seems to be that you are checking for the messages.

Reply:
@SanHalacogluImproving True, but we decided to stop adding tests like this since the message can change in the future and instead just check the error type
we didn't remove those tests but we can do that :)